source/fab/source_tree.py

```diff
@@ -57,16 +58,12 @@ def visit(self, candidate: Path) -> List[Path]:
                 message = \
                     f'Unhandled class "{task_class}" in extension map.'
                 raise TypeError(message)
-            # TODO: Make SQLite connection multiprocess safe
-            # self._queue.add_to_queue(task)
-            task.run()
+
+            self._queue.add_to_queue(task)
+            self._queue.add_to_queue(HashCalculator(hasher, self._state))
```

Comment:
One advantage of the hasher as a separate task is that you don't have to worry about any previous actions performed by the task stack altering the file you're hashing.